modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/MetadataComponentModelValidator.java

```diff
@@ -350,4 +351,8 @@ private boolean shouldValidateComponentOutputMetadata(ConnectableComponentModel
         .orElse(true);
   }
 
+  private boolean isCompileTime(ExtensionModel extensionModel) {
```

Comment:
this looks like something we would do (or we potentially are doing) in several places. Move this to `org.mule.runtime.module.extension.api.util.MuleExtensionUtils` and check for other places where this code is repeated.